### PR TITLE
Add bulk method, revamp documentation structure

### PR DESCRIPTION
**Pull request by akamor**

### CODE REVIEW

**Review by joeferraratonic — DISMISSED** [review]
I pushed a commit that changes the `de_identify_results` for `redact_bulk` from `List[Replacement]` where `Replacement` has an `idx` field to `List[List[Replacement]]` so that `de_identify_results`, `bulk_text`, and `bulk_redacted_text` are all lists of the same length. With this change I also removed the now unneeded `idx` field of `Replacement`. I also tested this change.

Otherwise, looks good to me.

File: docs/source/redact/redacting_text.rst [hunk 1]
```diff
@@ -28,97 +25,94 @@ This produces the following output:
...
 
     textual = TextualNer()
 
-    raw_synthesis = textual.llm_synthesis("My name is John, and today I am demoing Textual, a software product created by Tonic")
-    print(raw_synthesis.describe())
+    raw_redaction = textual.redact_bulk(["Tonic was founded in 2018", "John Smith is a person"])
+    print(raw_redaction.describe())
 
-This produces the following output:
+This produces the following output.  Note that the 'idx' property denotes the position in the original string list to which the result pertains.
```
Comment (gandersteele) [inline docs/source/redact/redacting_text.rst:66]:
```suggestion
This produces the following output:
```

File: tonic_textual/redact_api.py [hunk 2]
```diff
@@ -293,6 +294,87 @@ def redact(
...
+    
+    def redact_bulk(
+        self,
+        strings: List[str],
+        generator_config: Dict[str, PiiState] = dict(),
+        generator_default: PiiState = PiiState.Redaction,
+        random_seed: Optional[int] = None,
+        label_block_lists: Optional[Dict[str, List[str]]] = None,
+        label_allow_lists: Optional[Dict[str, List[str]]] = None,
+    ) -> RedactionResponse:
```
Comment (gandersteele) [inline tonic_textual/redact_api.py:306]:
```suggestion
    ) -> BulkRedactionResponse:
```

**Review by gandersteele — APPROVED** [review]
LGTM

**Pull request merged**